git_machete/tests/functional/test_machete.py

```diff
@@ -1381,3 +1552,235 @@ def test_update_with_fork_point_specified(self) -> None:
+        self.launch_command("github", "create-pr")
+        # ahead of origin state, push is advised and accepted
+        self.assert_command(
+            ['status'],
+            """
+            master
+            |
+            o-hotfix/add-trigger (diverged from origin)
+              |
+              o-ignore-trailing (diverged from & older than origin)
+                |
+                o-chore/fields (untracked)
+
+            develop
+            |
+            x-allow-ownership-link (ahead of origin)
+            | |
+            | x-build-chain (untracked)
+            |
+            o-call-ws *  PR #4
+              |
+              x-drop-constraint (untracked)
+            """,
+        )
+        self.repo_sandbox.check_out('chore/fields')
+        #  untracked state (can only create pr when branch is pushed)
+        self.launch_command("github", "create-pr")
+        self.assert_command(
+            ['status'],
+            """
+            master
+            |
+            o-hotfix/add-trigger (diverged from origin)
+              |
+              o-ignore-trailing (diverged from & older than origin)
+                |
+                o-chore/fields *  PR #5
+
+            develop
+            |
+            x-allow-ownership-link (ahead of origin)
+            | |
+            | x-build-chain (untracked)
+            |
+            o-call-ws  PR #4
+              |
+              x-drop-constraint (untracked)
+            """,
+        )
+
+        (
+            self.repo_sandbox.check_out('hotfix/add-trigger')
+            .commit('trigger released')
+            .commit('minor changes applied')
+        )
+
+        # diverged from and newer than origin
+        self.launch_command("github", "create-pr")
+        self.assert_command(
+            ['status'],
+            """
+            master
+            |
+            o-hotfix/add-trigger *  PR #6
+              |
+              x-ignore-trailing (diverged from & older than origin)
+                |
+                o-chore/fields  PR #5
+
+            develop
+            |
+            x-allow-ownership-link (ahead of origin)
+            | |
+            | x-build-chain (untracked)
+            |
+            o-call-ws  PR #4
+              |
+              x-drop-constraint (untracked)
+            """,
+        )
+        # check against attempt to create already existing pull request
+        machete_client = cmd.MacheteClient(cli_opts, git)
+        expected_error_message = "Pull request for branch hotfix/add-trigger is already created under link www.github.com!\nPR details: PR #6 by githubuser: hotfix/add-trigger -> master"
+        try:
+            machete_client.create_github_pr('hotfix/add-trigger', draft=False)
+        except MacheteException as e:
+            if e.parameter != expected_error_message:
+                self.fail(f'Actual Exception message: {e} \nis not equal to expected message: {expected_error_message}')
```

Comment:
This checks if error was raised, so if it didn't it will fail, which is desired.
If error was raised, so e is not None, we check that msg is as expected. If no error was raised this will be None so we can skip it (and we should, because None does not have argument 'parameter')
```suggestion
with self.assertRaises(MacheteException, ) as e:  
    machete_client.create_github_pr(....)
 if e:
     self.assertEqual(e.parameter, expected_error_message, 'Verify that expected error message has appeared when <insert_valid_conditions_of_the_tests>'). 

```

Reply:
yes, you are right, good spot :)